ai2thor/controller.py

```diff
@@ -481,19 +482,44 @@ def __init__(
                     DeprecationWarning,
                 )
 
+            # Let's set the scene for them!
+            if scene is None:
+                scenes_in_build = self.scenes_in_build
+                if not scenes_in_build:
+                    raise RuntimeError("No scenes are in your build of AI2-THOR!")
+
+                # we use FloorPlan28 because it is used for all the testing actions
+                if "FloorPlan28" in scenes_in_build:
+                    scene = "FloorPlan28"
+                else:
+                    # use a robothor scene
+                    robothor_scenes = set(self.robothor_scenes())
+
+                    # prioritize robothor if locobot is being used
+                    robothor_scenes_in_build = robothor_scenes.intersection(scenes_in_build)
+
+                    if (
+                        "locobot" == unity_initialization_parameters.get("agentMode", "default").lower() and
+                        robothor_scenes_in_build
+                    ):
+                        scene = random.choice(list(robothor_scenes_in_build))
+                    else:
+                        ithor_scenes = set(self.ithor_scenes())
+                        ithor_scenes_in_build = ithor_scenes.intersection(scenes_in_build)
+                        if ithor_scenes_in_build:
+                            # Prioritize iTHOR because that's what the default agent best uses.
+                            scene = random.choice(list(ithor_scenes_in_build))
+                        else:
+                            # perhaps only using RoboTHOR or using only custom scenes
+                            scene = random.choice(list(scenes_in_build))
+
             event = self.reset(scene)
-            if event.metadata["lastActionSuccess"]:
-                init_return = event.metadata["actionReturn"]
-                # older builds don't send actionReturn on Initialize
-                if init_return:
-                    self.server.set_init_params(init_return)
-                    logging.info("Initialize return: {}".format(init_return))
-            else:
-                raise RuntimeError(
-                    "Initialize action failure: {}".format(
-                        event.metadata["errorMessage"]
-                    )
-                )
+
+            # older builds don't send actionReturn on Initialize
+            init_return = event.metadata['actionReturn']
+            if init_return:
+                self.server.set_init_params(init_return)
+                logging.info("Initialize return: {}".format(init_return))
```

Comment:
I don't quite understand why the other logic was deleted. It seems like checking whether or not the reset succeeded (and otherwise throwing an error) seems important.

Reply:
The deleted logic was moved inside of `reset`. For reference `reset` now calls Unity's `Initialize`.

There should basically never be a reason that `Initialize` fails, and if it does, something has gone badly wrong.